From 9b1af7d4e4ebf7b5462edeb956b0651b6bf1178b Mon Sep 17 00:00:00 2001
From: Rui Matos <tiagomatos@gmail.com>
Date: Thu, 6 Oct 2016 15:57:01 +0200
Subject: [PATCH] MetaMonitorManagerKms: stop accounting for mode flags in
 refresh rates

This isn't technically needed and, in fact, makes us default to
interlaced modes in some cases which isn't desirable.

Note that X doesn't account for these flags either for its mode
refresh rates.

https://bugzilla.gnome.org/show_bug.cgi?id=772176
---
 src/backends/native/meta-monitor-manager-kms.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/backends/native/meta-monitor-manager-kms.c b/src/backends/native/meta-monitor-manager-kms.c
index 9215c1a..9e3fad2 100644
--- a/src/backends/native/meta-monitor-manager-kms.c
+++ b/src/backends/native/meta-monitor-manager-kms.c
@@ -423,10 +423,6 @@ drm_mode_vrefresh (const drmModeModeInfo *mode)
       refresh = (mode->clock * 1000000LL) / mode->htotal;
       refresh += (mode->vtotal / 2);
       refresh /= mode->vtotal;
-      if (mode->flags & DRM_MODE_FLAG_INTERLACE)
-        refresh *= 2;
-      if (mode->flags & DRM_MODE_FLAG_DBLSCAN)
-        refresh /= 2;
       if (mode->vscan > 1)
         refresh /= mode->vscan;
       refresh /= 1000.0;
-- 
2.9.3